Assets/MRTK/Core/Definitions/MixedRealityExperienceSettingsProfile.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.Tests.EditModeTests")]
+[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.Tests.PlayModeTests")]
+namespace Microsoft.MixedReality.Toolkit
+{
+    /// <summary>
+    /// Experience settings profile for the Mixed Reality Toolkit.
+    /// </summary>
+    [CreateAssetMenu(menuName = "Mixed Reality Toolkit/Profiles/Mixed Reality Toolkit Experience Settings Profile", fileName = "MixedRealityToolkitExperienceSettingsProfile", order = (int)CreateProfileMenuItemIndices.Configuration)]
+    [HelpURL("https://docs.microsoft.com/windows/mixed-reality/mrtk-unity/configuration/mixed-reality-configuration-guide")]
+    public class MixedRealityExperienceSettingsProfile : BaseMixedRealityProfile
+    {
+        [SerializeField]
+        [Tooltip("The scale of the Mixed Reality experience.")]
+        private ExperienceScale targetExperienceScale = ExperienceScale.Room;
+
+        /// <summary>
+        /// The desired the scale of the experience.
+        /// </summary>
+        public ExperienceScale TargetExperienceScale
+        {
+            get { return targetExperienceScale; }
+            set { targetExperienceScale = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("The height above the floor for the Mixed Reality Experience. 1 unit = 1 meter")]
+        private float floorHeight = 0.0f;
+
+        /// <summary>
+        /// The height above the floor for the Mixed Reality Experience. 1 unit = 1 meter
+        /// </summary>
+        public float FloorHeight
+        {
+            get { return floorHeight; }
+            set { floorHeight = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("If this setting is enabled, MRTK will position the camera Floor Height Units above the world origin to facilitate navigation. Set this to true only if the platform does not provide it's own floor height (WMR, Oculus etc.)")]
+        private bool alignCameraToFloorHeight = false;
+
+        /// <summary>
+        /// If this setting is enabled, MRTK will position the camera Floor Height Units above the world origin to facilitate navigation. Set this to true only if the platform does not provide it's own floor height (WMR, Oculus etc.)
+        /// </summary>
+        public bool AlignCameraToFloorHeight
```

Comment:
>Set this to true only if the platform does not provide it's own floor height

This setting still confuses me a bit, since I might not know all the platforms I plan to build for and I might build for platforms that both do and don't set their own floor height. It almost feels like we should key off whether the underlying platform was able to set the tracking space type by reacting to the line where we try to set it. Currently we just log:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/207ea221e4638c5697a3b5d618b36dccaded996f/Assets/MRTK/Providers/XRSDK/XRSDKBoundarySystem.cs#L112-L115

and

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/207ea221e4638c5697a3b5d618b36dccaded996f/Assets/MRTK/Services/BoundarySystem/XR2018/MixedRealityBoundarySystem.cs#L89-L92

Reply:
At the moment we only seem to set the tracking space /origin mode within the boundary system. Do we want to also do this check at the camera system level? (since that's and the scene content object itself are the only other places which account for floor height)